keps/sig-node/1287-in-place-update-pod-resources/README.md

```diff
@@ -435,6 +436,23 @@ Changes are always propogated through these 4 resource states in order:
 Desired --> Allocated --> Acknowledged --> Actual
 ```
 
+### Priority of Resize Requests
+
+Resize requests that come in will be added to a queue of pending resizes. Resize
+requests will be attempted according to the following priority:
+
+1. *Resource requests are not increasing*: Resizes that don't increase requests will be
+prioritized first. These resizes are expected to always succeed and would not be marked as
+pending.
+2. *PriorityClass*: Pods with a higher PriorityClass.
+3. *QoS Class*: Pods with a higher QoS class, where Guaranteed > Burstable > Best Effort.
+4. *Time since resize request*: If all else is the same, resizes that have been pending
+longer will be retried first (leveraging LastTransitionTime on the PodResizePending condition).
+
+A higher priority resize being marked as pending should not block the remaining pending resizes
+from being attempted, i.e. we will try all remaining resizes in the queue even if one is unsuccessful.
```

Comment:
This is the current behavior (and that is fine without more  data). But it is perhaps worth exploring if it is easy to add a mode where pending resizes in higher priority grade may block lower grade resizes. This may help avoid lower priority pods from gaming the system requesting small resizes while a bigger resize that is of higher priority starves.

Reply:
>  But it is perhaps worth exploring if it is easy to add a mode where pending resizes in higher priority grade may block lower grade resizes. This may help avoid lower priority pods from gaming the system requesting small resizes while a bigger resize that is of higher priority starves.

@dchen1107 said she had some thoughts around this. Also cc @tallclair for his thoughts.

My opinion - this is an interesting idea and worth discussing, but I think maybe not for 1.34. This might need to be part of a larger discussion about resizing and scheduler preemption. For example, if we move the resize decision to the scheduler and allow the resize of higher priority pods to preempt lower-priority pods, would there still be any reason to have this?

Reply:
Yeah, I agree with @natasha41575 on this. There is a lot we _could_ do here, but I don't want the Kubelet to be making complicated optimization decisions around this. If we want to have more complex logic about interplay between workloads, I'd rather push that decision up into the scheduler (which we might want to do anyway).

Reply:
SG.  Just making sure we capture scope accurately. Iterating in smaller steps is always the way to go!

Reply:
+1 on both @tallclair and @natasha41575 above. There is a separate ongoing discussion on resizing and scheduling preemption, we can enhance the feature later.